src/main/java/uk/gov/pay/ledger/transaction/service/TransactionMetadataService.java

```diff
@@ -35,25 +32,19 @@ public TransactionMetadataService(MetadataKeyDao metadataKeyDao,
         this.transactionDao = transactionDao;
     }
 
-    public void upsertMetadataFor(Event event) {
-        JsonNode eventDataNode;
-        try {
-            eventDataNode = new ObjectMapper().readTree(event.getEventData());
-        } catch (JsonProcessingException e) {
-            LOGGER.error("Unable to parse incoming event payload: {}", e.getMessage());
-            return;
-        }
-        if (eventDataNode != null && eventDataNode.has("external_metadata")) {
-            transactionDao.findTransactionByExternalId(event.getResourceExternalId())
+    public void upsertMetadataFor(EventDigest eventDigest) {
+        var eventData = eventDigest.getEventPayload();
+
+        transactionDao.findTransactionByExternalId(eventDigest.getResourceExternalId())
                     .ifPresent(transactionEntity -> {
-                        eventDataNode.get("external_metadata").fields().forEachRemaining(metadata -> {
-                            metadataKeyDao.insertIfNotExist(metadata.getKey());
-                            String value = metadata.getValue().textValue();
-                            transactionMetadataDao
-                                    .upsert(transactionEntity.getId(), metadata.getKey(), value);
-                        });
-                    });
-        }
+                        if (eventData.containsKey("external_metadata")) {
+                            var metadata = (Map<String, Object>) eventData.get("external_metadata");
+                            metadata.keySet().forEach(key -> {
```

Comment:
Would this be better using `metadata.entrySet()` to match what we were doing before?